### PR TITLE
Feature/new app layout

### PR DESCRIPTION
**Pull request by jegasriskantha**
relates to relates to https://github.com/IPG-Automotive-UK/cloud-licensing-web/issues/274

## Changes

- Adds a new generic layout that will be used in the cloud licensing app

- The generic layout builds on the VIRTO app layout but uses the newly created generic app header and has a sidebar width that is designed to be used with the the stacked variant of the sidebar item. 

- When trying to configure the app layout for the cloud licensing app I noticed an issue with the sidebarItem types and some styling when the sidebarItems are stacked, I have made updates to both these here. 
  - iconStyle and TextStyle now have the SxProps type not the CSSProperties type. This is more appropriate and allows for things like this `  textStyle={{ ".MuiTypography-root": { fontSize: "12px" } }}` which CSSProperties does not. 
  - I removed the horizontal padding and added padding vertically on the ListItemButton when the items are "stacked" to align with the styling in the Figma prototype for Cloud Licensing. 
  - Based on some feedback from @Sowbhagya-ipg https://github.com/IPG-Automotive-UK/react-ui/pull/604#issuecomment-1545361086 I also removed some left padding that was the appHeader. Fixed here:  https://github.com/IPG-Automotive-UK/react-ui/pull/604/commits/882c9f7124d1f7c021265a89247b661c6743ae5e
  - I have also updated the badge to only be a dot when the sidebar items are stacked. This was also confirmed with @Sowbhagya-ipg 

### Do Not Merge
This PR builds of work done here: https://github.com/IPG-Automotive-UK/react-ui/pull/602 and should not be merged till this is merged.

## UI/UX
This is the default app layout
image

This is what the app layout looks like configured for cloud licensing as per the figma prototype
Updated screenshot after review with @Sowbhagya-ipg 
image

This is the Figma prototype:
https://www.figma.com/proto/O3IAobLTPcjIO1WaxaPod6/IPG-Cloud-Licensing?node-id=68-14265&starting-point-node-id=68%3A14265
image

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [ ] Branch has been run in docker.
- [x] Assigned to me.
- [x] Relevant tags added.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

**Comment by Sowbhagya-ipg** [comment]
The app bar margin on the left and right doesn't seem equal. @jegasriskantha  Please check.

**Comment by Sowbhagya-ipg** [comment]
Checked with @jegasriskantha. Look's Good

**Comment by jegasriskantha** [comment]
Apologies @nikitharani  I had missed some exports, they have been updated now.

File: src/AppLayout/AppLayout.types.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   * ```
+   * function(event: object) => void
+   * ```
+   * _event_: The event source of the callback.
+   */
+  onChangePassword: (event: React.MouseEvent<HTMLElement>) => void;
+  /**
+   * Callback fired when the user clicks on "Logout".
+   *
+   * **Signature**
```
Comment (nikitharani) [inline src/AppLayout/AppLayout.types.ts:40]: we can remove signatures

**Review by nikitharani — CHANGES_REQUESTED** [review]
Just a small change requested otherwise LGTM.

**Pull request merged**